spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/reload/EventBasedConfigurationChangeDetector.java

```diff
@@ -138,6 +117,46 @@ public void unwatch() {
 		}
 	}
 
+	private void watchConfigMap(final String name) {
+		Watcher<ConfigMap> watcher = new Watcher<ConfigMap>() {
+			@Override
+			public void eventReceived(Action action, ConfigMap configMap) {
+				onEvent(configMap);
+			}
+
+			@Override
+			public void onClose(KubernetesClientException error) {
+				if (error != null) {
+					if (log.isDebugEnabled()) {
+						log.warn("Try to re-watch: " + name, error);
+					}
+					watchConfigMap(name);
```

Comment:
Could we not get into some kind of infinite loop where we indefinitely retry here?  It would be nice if we could add some backoff functionality or even just try a set number of times.

Reply:
Ok. I'll try to implement backoff because the errors reported in #558 occur periodically.

Reply:
I added backoff functionality by ba47a20. Please check it.

Reply:
@ryanjbaxter Should I make any additional changes?